docs/architecture-decision-records/2023-01-initial-architecture.md

```diff
@@ -0,0 +1,240 @@
+
+Issues:
+
+- Apple M1 architecture has Tier 3 support only (not production ready).
+
+- It's not clear whether functions exported by WASM modules can be async. This
+  may require custom glue code to be written.
+
+- There is no stable API for HTTP/HTTPS. WASI does not support opening new
+  network connections yet, and API for crypto primitives required by TLS is not
+  specified yet.
+
+- Supports WASM only, it cannot run JavaScript/TypeScript code.
+
+### WasmEdge
+
+_https://wasmedge.org/_
+
+WasmEdge aims to bring the cloud-native and serverless application paradigms to
+Edge Computing. It's in many aspects similar to Wasmtime. Unfortunately, it's a
+C/C++ project, which makes it difficult to integrate into Rust projects,
+especially on Apple Silicon, where binaries must be signed.
+
+### Wasmer
+
+_https://wasmer.io_
+
+Wasmer is similar to Wasmtime.
+
+Benefits:
+
+- WebAssembly package manager offering a catalogue of WASM modules
+
+- Supports WASI (filesystem)
+
+- Popular in Web3 space (ChainSafe, Fluence, Hyperledger, and more)
+
+Issues:
+
+- No support for an asynchronous programming model & non-blocking I/A
+
+- Lack of documentation for advanced users. For example, I could not figure out
+  how to configure an allow-list of directories allowed to access from WASI.
+
+- Instead of contributing to existing bindgen initiatives, they started a new
+  project [fp-bindgen](https://github.com/fiberplane/fp-bindgen) that works with
+  Wasmer only.
+
+- The project is owned and run by a single company - Wasmer, Inc.
+
+## Decision
+
+<!--
+What is the change that we're proposing and/or doing?
+-->
+
+**I am proposing to use Wasmtime as the underlying engine for the Station
+Runtime.**
+
+While Deno would allow us to start faster, by providing Fetch API and Filesystem
+access with built-in sandboxing, I feel it would make it much harder for us to
+build past the built-in features (add IPFS & libp2p, implement resource usage
+limiting). Our requirements are unlikely to align with Denoland's vision for
+Deno, so we may have a hard time getting our changes accepted.
+
+We are going to build our Runtime incrementally. Early Module authors will
+likely need to contribute new features to our Runtime to enable their modules to
+be built. This will be easier if the interface between modules and the runtime
+is Rust-native. If we built on top of Deno with wasm-bindgen, js-sys and
+web-sys, then module authors would need to understand both Rust and JavaScript.
+
+That's why I believe it's better to choose a more bare-bone WASM engine.
+
+From the different WASM engine options out there, it seems that Wasmtime is best
+aligned with ProtocolLab's vision of driving technical breakthroughs in the
+open. The project is community-driven and contributes to other WASM initiatives.
+
+On the technical side, it is the only engine that supports the asynchronous
+programming model (AFAICT). It also has the best documentation.
```

Comment:
```suggestion
programming model (AFAICT). It also has the most extensive documentation.
```

I try to avoid words like "best", because they can mean different things to different people, and replace them with something more semantic. Does this work for you?